lib/run.js

```diff
@@ -11,12 +11,23 @@ const reInterval = require('reinterval')
 const histAsObj = histUtil.histAsObj
 const addPercentiles = histUtil.addPercentiles
 
-function run (opts, cb) {
+function run (opts, cb, tracker) {
   const cbPassedIn = (typeof cb === 'function')
 
   cb = cb || noop
 
-  const tracker = new EE()
+  tracker = tracker || new EE()
+
+  if (!cbPassedIn && !opts.forever) {
+    tracker.promise = new Promise((resolve, reject) => {
```

Comment:
do we need to keep a reference to the promise on the tracker itself? I would prefer not to.

Reply:
fixed